yt_dlp/extractor/detik.py

```diff
@@ -0,0 +1,127 @@
+            'title': 'Crash-crash Horor di Paruh Pertama MotoGP 2022',
+            'description': 'md5:fbcc6687572ad7d16eb521b76daa50e4',
+            'timestamp': 1656900391,
+            'duration': 107,
+            'tags': ['marc marquez', 'fabio quartararo', 'francesco bagnaia', 'motogp crash', 'motogp 2022'],
+            'upload_date': '20220704',
+        }
+    }, {
+        # adu-perspektif
+        'url': 'https://20.detik.com/adu-perspektif/20220518-220518144/24-tahun-reformasi-dan-alarm-demokrasi-dari-filipina',
+        'info_dict': {
+            'id': '220518144',
+            'ext': 'mp4',
+            'title': '24 Tahun Reformasi dan Alarm Demokrasi dari Filipina',
+            'upload_date': '20220518',
+            'timestamp': 1652888623,
+            'duration': 7385,
+            'tags': ['politik', 'adu perspektif', 'indonesia', 'filipina', 'demokrasi'],
+            'description': 'md5:8eaaf440b839c3d02dca8c9bbbb099a9',
+            'thumbnail': 'https://cdnv.detik.com/videoservice/AdminTV/2022/05/18/adpers_18_mei_compressed-20220518230458-custom.jpg?w=650&q=80',
+        }
+    }, {
+        # sosok
+        'url': 'https://20.detik.com/sosok/20220702-220703032/resa-boenard-si-princess-bantar-gebang',
+        'info_dict': {
+            'id': '220703032',
+            'ext': 'mp4',
+            'timestamp': 1656799238,
+            'thumbnail': 'https://cdnv.detik.com/videoservice/AdminTV/2022/07/02/SOSOK_BGBJ-20220702191138-custom.jpg?w=650&q=80',
+            'title': 'Resa Boenard Si \'Princess Bantar Gebang\'',
+            'description': 'md5:84ea66306a0285330de6a13fc6218b78',
+            'tags': ['sosok', 'sosok20d', 'bantar gebang', 'bgbj', 'resa boenard', 'bantar gebang bgbj', 'bgbj bantar gebang', 'sosok bantar gebang', 'sosok bgbj', 'bgbj resa boenard'],
+            'upload_date': '20220702',
+            'duration': 650,
+        }
+    }, {
+        # viral
+        'url': 'https://20.detik.com/viral/20220603-220603135/merasakan-bus-imut-tanpa-pengemudi-muter-muter-di-kawasan-bsd-city',
+        'info_dict': {
+            'id': '220603135',
+            'ext': 'mp4',
+            'description': 'md5:4771fe101aa303edb829c59c26f9e7c6',
+            'timestamp': 1654279105,
+            'title': 'Merasakan Bus Imut Tanpa Pengemudi, Muter-muter di Kawasan BSD City',
+            'tags': ['viral', 'autonomous vehicle', 'electric', 'shuttle bus'],
+            'thumbnail': 'https://cdnv.detik.com/videoservice/AdminTV/2022/06/03/VIRAL_BUS_NO_SUPIR-20220604004707-custom.jpg?w=650&q=80',
+            'duration': 593,
+            'upload_date': '20220603'
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        json_ld_data = list(self._yield_json_ld(webpage, display_id))[0]
```

Comment:
Any reason for not using _search_json_ld directly?

Reply:
I was not sure what the json ld data gave when there is more than one json ld, so i set it based on the order. After testing with `_search_json_ld`, it works as intended, so i changed this line to `_search_json_ld`